tests/scenario/test_cert_handler/test_cert_handler_v1.py

```diff
@@ -183,48 +191,53 @@ def test_cert_joins_peer_vault_backend(ctx_juju2, certificates, leader):
         assert mgr.charm.ch.private_key
 
 
-def test_renew_csr_on_sans_change(ctx, certificates):
-    # generate a CSR
+@pytest.mark.parametrize(
+    "event,generate_call_count",
+    (("update_status", 0), ("start", 0), ("install", 0), ("config_changed", 1)),
```

Comment:
it's not entirely clear to me why we expect config_changed to result in a generate_cert call, but not the other events.

Reply:
because CertHandler generates the cert on the `config_changed` event https://github.com/canonical/observability-libs/blob/main/lib/charms/observability_libs/v1/cert_handler.py#L472

Reply:
will add a comment with that on the test